python/shark_turbine/kernel/compiler/builder.py

```diff
@@ -124,44 +161,73 @@ def promote_index_to_f32(self, value: Value, to_type: IrType) -> Value:
-    def binary_add_index_index(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.addi(lhs, rhs)
+    def binary_add_index_index(
+        self, lhs: IRProxyValue, rhs: IRProxyValue
+    ) -> IRProxyValue:
+        return IRProxyValue(arith_d.addi(lhs.ir_value, rhs.ir_value))
+
+    def binary_mul_index_index(
+        self, lhs: IRProxyValue, rhs: IRProxyValue
+    ) -> IRProxyValue:
+        return IRProxyValue(arith_d.muli(lhs.ir_value, rhs.ir_value))
+
+    def binary_sub_index_index(
+        self, lhs: IRProxyValue, rhs: IRProxyValue
+    ) -> IRProxyValue:
+        return IRProxyValue(arith_d.subi(lhs.ir_value, rhs.ir_value))
+
+    def binary_mod_index_index(
+        self, lhs: IRProxyValue, rhs: IRProxyValue
+    ) -> IRProxyValue:
+        return IRProxyValue(arith_d.remsi(lhs.ir_value, rhs.ir_value))
+
+    def binary_floordiv_index_index(
+        self, lhs: IRProxyValue, rhs: IRProxyValue
+    ) -> IRProxyValue:
+        return IRProxyValue(arith_d.floordivsi(lhs.ir_value, rhs.ir_value))
 
-    def binary_mul_index_index(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.muli(lhs, rhs)
+    # Binary f32 arithmetic
+    def binary_add_f32_f32(self, lhs: IRProxyValue, rhs: IRProxyValue) -> IRProxyValue:
+        return IRProxyValue(arith_d.addf(lhs.ir_value, rhs.ir_value))
 
-    def binary_sub_index_index(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.subi(lhs, rhs)
+    def binary_mul_f32_f32(self, lhs: IRProxyValue, rhs: IRProxyValue) -> IRProxyValue:
+        return IRProxyValue(arith_d.mulf(lhs.ir_value, rhs.ir_value))
 
-    def binary_mod_index_index(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.remsi(lhs, rhs)
+    def binary_sub_f32_f32(self, lhs: IRProxyValue, rhs: IRProxyValue) -> IRProxyValue:
+        return IRProxyValue(arith_d.subf(lhs.ir_value, rhs.ir_value))
 
-    def binary_floordiv_index_index(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.floordivsi(lhs, rhs)
+    def binary_mod_f32_f32(self, lhs: IRProxyValue, rhs: IRProxyValue) -> IRProxyValue:
+        return IRProxyValue(arith_d.remf(lhs.ir_value, rhs.ir_value))
 
-    # Binary f32 arithmetic
-    def binary_add_f32_f32(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.addf(lhs, rhs)
+    def binary_truediv_f32_f32(
+        self, lhs: IRProxyValue, rhs: IRProxyValue
+    ) -> IRProxyValue:
+        return IRProxyValue(arith_d.divf(lhs.ir_value, rhs.ir_value))
 
-    def binary_mul_f32_f32(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.mulf(lhs, rhs)
 
-    def binary_sub_f32_f32(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.subf(lhs, rhs)
-
-    def binary_mod_f32_f32(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.remf(lhs, rhs)
+ScalarBuilder = _ScalarBuilder()
 
-    def binary_truediv_f32_f32(self, lhs: Value, rhs: Value) -> Value:
-        return arith_d.divf(lhs, rhs)
 
+# class SliceSpecExtractor:
```

Comment:
Should we delete these?

Reply:
ah, missed that one in my pass through looking for random crap I left lying around.